tasks/4-deploy-ace-bar.yaml

```diff
@@ -0,0 +1,215 @@
+      echo "CHART: $CHART"
+
+      export CHART_NAME=$(cat $CHART | awk '/name:/ {print $2}')
+      echo "CHART_NAME: $CHART_NAME"
+
+      # Update Chart name and version
+      if [[ "${CHART_NAME}" != "${APP_NAME}" ]]; then
+        echo "Renaming chart from ${CHART_NAME} to ${APP_NAME}"
+        cp -R "${CHART_ROOT}/${CHART_NAME}" "${CHART_ROOT}/${APP_NAME}"
+        cat "${CHART_ROOT}/${CHART_NAME}/Chart.yaml" | \
+            yq w - name "${APP_NAME}" | \
+            yq w - version "1.0.0-rc${IMAGE_VERSION}" > "${CHART_ROOT}/${APP_NAME}/Chart.yaml"
+      else
+        echo "Chart name and image name match: ${APP_NAME}"
+      fi
+
+      CHART_PATH="${CHART_ROOT}/${APP_NAME}"
+
+      RELEASE_NAME="${APP_NAME}"
+      echo "RELEASE_NAME: $RELEASE_NAME"
+
+      echo $(helm version)
+
+      echo "CHECKING CHART (lint)"
+      helm lint ${CHART_PATH}
+
+      PREFIX=""
+      if [[ -f "${CHART_PATH}/requirements.yaml" ]] || grep -Eq "^dependencies:" "${CHART_PATH}/Chart.yaml"; then
+          DEPENDENCY_FILE="${CHART_PATH}/Chart.yaml"
+          if [[ -f "${CHART_PATH}/requirements.yaml" ]]; then
+              DEPENDENCY_FILE="${CHART_PATH}/requirements.yaml"
+          fi
+
+          PREFIX="$(yq r -j "${DEPENDENCY_FILE}" | jq -r '.dependencies | .[] | .alias // .name' | head -1)."
+      fi
+
+      # Update helm chart with repository and tag values
+      cat ${CHART_PATH}/values.yaml | \
+          yq w - nameOverride "${APP_NAME}" | \
+          yq w - fullnameOverride "${APP_NAME}" | \
+          yq w - vcsInfo.repoUrl "${GIT_URL}" | \
+          yq w - vcsInfo.branch "${GIT_REVISION}" | \
+          yq w - image.repository "${IMAGE_URL}" | \
+          yq w - --style single image.tag "${IMAGE_VERSION}" | \
+          yq w - ingress.enabled "${INGRESS_ENABLED}" | \
+          yq w - route.enabled "${ROUTE_ENABLED}" > ./values.yaml.tmp
+      cp ./values.yaml.tmp ${CHART_PATH}/values.yaml
+      cat ${CHART_PATH}/values.yaml
+
+      helm dep update "${CHART_PATH}"
+
+      # Using 'upgrade --install" for rolling updates. Note that subsequent updates will occur in the same namespace the release is currently deployed in, ignoring the explicit--namespace argument".
+      helm template ${RELEASE_NAME} ${CHART_PATH} \
+          --namespace ${APP_NAMESPACE} \
+          --set ingress.tlsSecretName="${TLS_SECRET_NAME}" \
+          --set ingress.subdomain="${INGRESS_SUBDOMAIN}" > ./release.yaml
+
+      echo -e "Generated release yaml for: ${APP_NAMESPACE}/${APP_NAME}"
+      cat ./release.yaml
+
+      echo -e "Deploying into: ${APP_NAMESPACE}/${APP_NAME}"
+      kubectl apply -n ${APP_NAMESPACE} -f ./release.yaml --validate=false
+  - name: deployment-check
+    image: $(params.tools-image)
+    script: |
+      #!/usr/bin/env bash
+      set -x
+
+      DEPLOYMENT_NAME=$(oc get deployments | grep -v "el-" | grep $(params.app-name) | cut -f1 -d ' ')
+
+      while [[ -z "${DEPLOYMENT_NAME}" ]]; do
```

Comment:
We don't need this task, move this wait for deployment to exist to the original deploy task 
here this line https://github.com/IBM/ibm-garage-tekton-tasks/blob/main/tasks/4-deploy.yaml#L184

Reply:
Consolidated changes to deploy task in latest commit.